language-support/hs/bindings/src/DA/Ledger/Services/MeteringReportService.hs

```diff
@@ -0,0 +1,116 @@
+-- Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+{-# LANGUAGE DuplicateRecordFields #-}
+
+module DA.Ledger.Services.MeteringReportService (
+    getMeteringReport, 
+    MeteringReport(..), 
+    isoTimeToTimestamp,
+  ) where
+
+
+import Data.Aeson ( KeyValue((.=)), ToJSON(..), object)
+import DA.Ledger.Convert
+import DA.Ledger.GrpcWrapUtils
+import DA.Ledger.LedgerService
+import DA.Ledger.Types
+import qualified Data.Text.Lazy as TL
+import Network.GRPC.HighLevel.Generated
+import qualified Com.Daml.Ledger.Api.V1.Admin.MeteringReportService as LL
+import Data.Maybe (maybeToList)
+import qualified Data.Time.Clock.System as System
+import qualified Data.Time.Format.ISO8601 as ISO8601
+import GHC.Int (Int64)
+import GHC.Word (Word32)
+
+data MeteredApplication = MeteredApplication {
+  application :: ApplicationId
+, events :: Int64
+} deriving (Show)
+
+instance ToJSON MeteredApplication where
+  toJSON (MeteredApplication application events) =
+    object
+    [   "application" .= unApplicationId application
+    ,   "events" .= events
+    ]
+
+data MeteringReport = MeteringReport {
+  participant :: ParticipantId
+, from  :: Timestamp
+, toRequested :: Maybe Timestamp
+, toActual :: Timestamp
+, applications :: [MeteredApplication]
+} deriving (Show)
+
+instance ToJSON MeteringReport where
+  toJSON (MeteringReport participant from toRequested toActual applications) =
+    object (
+    [   "participant" .= unParticipantId participant
+    ,   "from" .= timestampToIso8601 from
+    ,   "to-actual" .= timestampToIso8601 toActual
```

Comment:
```suggestion
    ,   "to_actual" .= timestampToIso8601 toActual
```
Not sure if there is a specific reason for the hyphen but ime underscores are far more common in JSON encodings so I’d suggest to use that for ledger metering as well.

Reply:
Will update provided all are happy with the spec change.